hawkular-inventory-impl-cassandra/src/test/java/org/hawkular/inventory/impl/cassandra/CassandraBackendTest.java

```diff
@@ -0,0 +1,64 @@
+package org.hawkular.inventory.impl.cassandra;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.datastax.driver.core.Row;
+import com.google.common.collect.Lists;
+
+/**
+ * @author Joel Takvorian
+ */
+public class CassandraBackendTest {
+
+    @Test
+    public void shouldProcessTransitiveClosure() {
+        List<Row> relationships = Lists.newArrayList(
```

Comment:
What's the performance characteristics of this with say 20000 relationships processed where only let's say 20 are in the resulting closure?